udata/features/webhooks/triggers.py

```diff
@@ -0,0 +1,79 @@
+import json
+
+from udata.core.discussions.signals import (
+    on_new_discussion, on_new_discussion_comment, on_discussion_closed,
+)
+
+from udata.features.webhooks.tasks import dispatch
+from udata.models import Dataset, Organization, Reuse, CommunityResource
+
+
+@Dataset.on_create.connect
+def on_dataset_create(dataset):
+    if not dataset.private:
+        dispatch('datagouvfr.dataset.created', dataset.to_json())
+
+
+@Dataset.on_delete.connect
+def on_dataset_delete(dataset):
+    dispatch('datagouvfr.dataset.deleted', dataset.to_json())
+
+
+@Dataset.on_update.connect
+def on_dataset_update(dataset):
+    updates, _ = dataset._delta()
+    if 'private' in updates and not dataset.private:
+        dispatch('datagouvfr.dataset.created', dataset.to_json())
+    else:
+        dispatch('datagouvfr.dataset.updated', dataset.to_json())
+
+
+@on_new_discussion.connect
+def on_new_discussion(discussion):
```

Comment:
Same question on discussion events, don't we want to add the `if not dataset.private` condition on dataset deletion ?